src/utils/push.rs

```diff
@@ -0,0 +1,137 @@
+// SPDX-FileCopyrightText: 2020 Serokell <https://serokell.io/>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+use std::process::Stdio;
+use tokio::process::Command;
+
+pub async fn push_profile(
+    supports_flakes: bool,
+    check_sigs: bool,
+    repo: &str,
+    deploy_data: &super::DeployData<'_>,
+    deploy_defs: &super::DeployDefs<'_>,
+    keep_result: bool,
+    result_path: Option<&str>,
+) -> Result<(), Box<dyn std::error::Error>> {
+    info!(
+        "Building profile `{}` for node `{}`",
+        deploy_data.profile_name, deploy_data.node_name
+    );
+
+    let mut build_c = if supports_flakes {
+        Command::new("nix")
+    } else {
+        Command::new("nix-build")
+    };
+
+    let mut build_command = if supports_flakes {
+        build_c.arg("build").arg("--no-link").arg(format!(
+            "{}#deploy.nodes.{}.profiles.{}.path",
+            repo, deploy_data.node_name, deploy_data.profile_name
+        ))
+    } else {
+        build_c
+            .arg(&repo)
+            .arg("--no-out-link")
+            .arg("-A")
+            .arg(format!(
+                "deploy.nodes.{}.profiles.{}.path",
+                deploy_data.node_name, deploy_data.profile_name
+            ))
+    };
```

Comment:
Is there a reason for two separate if's? Seems less readable to me.

Reply:
Kindof? I can't let the Command itself go out of scope and this solved it